src/metabase/driver/util.clj

```diff
@@ -384,11 +384,15 @@
                 (assoc :visible-if v-ifs*))))
          final-props)))
 
+(def ^:private data-url-pattern #"^data:[^;]+;base64,")
+
 (defn decode-uploaded
   "Decode `uploaded-data` as an uploaded field.
   Optionally strip the Base64 MIME prefix."
-  ^bytes [uploaded-data]
-  (u/decode-base64-to-bytes (str/replace uploaded-data #"^data:[^;]+;base64," "")))
+  ^bytes [^String uploaded-data]
+  (if (re-find data-url-pattern uploaded-data)
+    (u/decode-base64-to-bytes (str/replace uploaded-data data-url-pattern ""))
+    (.getBytes uploaded-data "UTF-8")))
```

Comment:
Now that I think about it, this could be a breaking change, if we previously stored some data in Base64 but without the `data:` prefix from the frontend. I don't know if that was the case at some point in the past.

Reply:
Since stripping the prefix was optional, I fully expect that we can have base64 without prefix. I wonder if treating all valid base64 strings as base64 is viable. It's not 100% right, but might be reliable enough if the strings are not too short.

Reply:
I'll leave `decode-uploaded` so that it always decodes base64, but add a `maybe-decode-uploaded`, just for Snowflake in this case.